packages/lit-dev-content/site/docs/v3/releases/upgrade.md

```diff
@@ -95,10 +112,52 @@ Experimental hydration support has been moved out of core libraries and into
 
 ```ts
 // Removed
-import 'lit/experimental-hydrate-support.js'
+import 'lit/experimental-hydrate-support.js';
 import {hydrate} from 'lit/experimental-hydrate.js';
 
 // Updated
 import '@lit-labs/ssr-client/lit-element-hydrate-support.js';
 import {hydrate} from '@lit-labs/ssr-client';
 ```
+
+### Optional: To aid migration to standard decorators, decorate hand written setters {#decorated-getter}
+
+When using standard decorators, `@property()` and `@state()` must be used on the
+setter for hand-written accessors, and will throw a type error when applied on a
+hand written getter. Thus we recommend incrementally migrating.
+
+```ts
+// This will throw a type error if using standard decorators.
+@property()
+get myProperty () { ... }
+
+set myProperty (val) { ... }
+
+// Becomes:
+get myProperty () { ... }
+
+@property()
+set myProperty (val) { ... }
+```
+
+### Optional: upgrade to standard decorators {#standard-decorator-migration}
+
+To use standard decorators, your decorators should add the `accessor` keyword.
+Lit 3 decorators are flexible and work as both experimental decorators, and as
+standard decorators.
+
+```ts
+class MyElement extends LitElement {
+  // Lit 3.0 experimental decorators, which are backwards compatible with Lit 2.0
+  @property()
+  myProperty = "initial value"
+
+  // Lit 3.0 adds support for standard decorators.
+  // When using TypeScript:
+  //  - `accessor` keyword is optional when `experimentalDecorators: true`.
+  //  - `accessor` keyword is required when `experimentalDecorators: false`.
+  @property()
+  accessor myProperty = "initial value"
+...
+}
+```
```

Comment:
```suggestion
\```
#### Move decorators from getters to setters

Standard decorators can only replace the class member they're directly applied to. Lit decorators need to intercept property setting, so the decorators must be applied to setters. This is different from the Lit 2 recommendation of applying decorators to getters.

For `@property()` and `@state()` you should also remove any `this.requestUpdate()` calls in the setter since this is done automatically now. If you need to _not_ have `requestUpdate()` called, you'll have to use the `hasChanged` property option.

Note that for `@property()` and `@state()` the _getter_ will be called by the decorator when setting the property to retrieve the old value. This means that you _must_ define both a getter and a setter.

Before:
```ts
class MyElement extends LitElement {
  private _foo = 42;
  set(v) {
    const oldValue = this._foo;
    this._foo = v;
    this.requestUpdate('foo', oldValue);
  }
  @property()
  get() {
    return this._foo;
  }
}
\```

After:
```ts
class MyElement extends LitElement {
  private _foo = 42;
  @property()
  set(v) {
    this._foo = v;
  }
  get() {
    return this._foo;
  }
}
\```
```